src/debug_info.rs

```diff
@@ -1,30 +1,1060 @@
+        file: DIFile<'ctx>,
+        line_no: u32,
+        scope: DIScope<'ctx>,
+    ) -> DIDerivedType<'ctx> {
+        use llvm_sys::debuginfo::LLVMDIBuilderCreateTypedef;
+        let name = to_c_str(name);
+        let metadata_ref = unsafe {
+            LLVMDIBuilderCreateTypedef(
+                self.builder,
+                ditype.metadata_ref,
+                name.as_ptr(),
+                name.to_bytes().len(),
+                file.metadata_ref,
+                line_no,
+                scope.metadata_ref,
+            )
+        };
+        DIDerivedType {
+            metadata_ref,
+            _marker: PhantomData,
+        }
+    }
+
+    /// Create a typedef (alias) of `ditype`
+    #[llvm_versions(10.0..=latest)]
+    pub fn create_typedef(
+        &self,
+        ditype: DIType<'ctx>,
+        name: &str,
+        file: DIFile<'ctx>,
+        line_no: u32,
+        scope: DIScope<'ctx>,
+        align_in_bits: u32,
+    ) -> DIDerivedType<'ctx> {
+        use llvm_sys::debuginfo::LLVMDIBuilderCreateTypedef;
+        let name = to_c_str(name);
+        let metadata_ref = unsafe {
+            LLVMDIBuilderCreateTypedef(
+                self.builder,
+                ditype.metadata_ref,
+                name.as_ptr(),
+                name.to_bytes().len(),
+                file.metadata_ref,
+                line_no,
+                scope.metadata_ref,
+                align_in_bits,
+            )
+        };
+        DIDerivedType {
+            metadata_ref,
+            _marker: PhantomData,
+        }
+    }
+
+    /// Create union type of multiple types.
+    pub fn create_union_type(
+        &self,
+        scope: DIScope<'ctx>,
+        name: &str,
+        file: DIFile<'ctx>,
+        line_no: u32,
+        size_in_bits: u64,
+        align_in_bits: u32,
+        flags: DIFlags,
+        elements: Vec<DIType<'ctx>>,
+        runtime_language: u32,
+        unique_id: &str,
+    ) -> DICompositeType<'ctx> {
+        use llvm_sys::debuginfo::LLVMDIBuilderCreateUnionType;
+        let mut elements: Vec<LLVMMetadataRef> =
+            elements.into_iter().map(|dt| dt.metadata_ref).collect();
+        let name = to_c_str(name);
+        let unique_id = to_c_str(unique_id);
+        let metadata_ref = unsafe {
+            LLVMDIBuilderCreateUnionType(
+                self.builder,
+                scope.metadata_ref,
+                name.as_ptr(),
+                name.to_bytes().len(),
+                file.metadata_ref,
+                line_no,
+                size_in_bits,
+                align_in_bits,
+                flags.into(),
+                elements.as_mut_ptr(),
+                elements.len().try_into().unwrap(),
+                runtime_language,
+                unique_id.as_ptr(),
+                unique_id.to_bytes().len(),
+            )
+        };
+        DICompositeType {
+            metadata_ref,
+            _marker: PhantomData,
+        }
+    }
+
+    /// Create a function type
```

Comment:
```suggestion
    /// Create a subroutine type
```
Or if you like `subroutine (function)`? Depending on the source language, subroutines might or might not be functions.